azurepipelines/e2e_test/scenarios/test_runner/apt_deployment.py

```diff
@@ -48,6 +49,16 @@ def test_AptDeployment(self):
         self.aduTestConfig = DuAutomatedTestConfigurationManager.FromOSEnvironment()
         self.duTestHelper = self.aduTestConfig.CreateDeviceUpdateTestHelper()
 
+        self.aduScenarioDefinition = DuScenarioDefinitionManager.FromOSEnvironment()
+
+        test_device_id = self.aduScenarioDefinition.test_device_id
+        test_adu_group = self.aduScenarioDefinition.test_adu_group
+        global test_result_file_prefix
```

Comment:
Globals scare me. What's the use case for declaring it here within all the tests?

Reply:
I am not familiar with how exactly this works, it uses test_result_file_prefix outside the class. I am not sure what's the best way to do it?:
if (__name__ == "__main__"):
    #
    # Create the IO pipe
    #
    out = io.BytesIO()

    #
    # Exercise the TestCase and all the tests within it.
    #
    unittest.main(testRunner=xmlrunner.XMLTestRunner(output=out),
                  failfast=False, buffer=False, catchbreak=False, exit=False)

    #
    # Finally transform the output unto the X/JUnit XML file format
    #
    with open('./testresults/' + test_result_file_prefix + '-apt-deployment-test.xml', 'wb') as report:
        report.write(transform(out.getvalue()))